scylla-cdc/src/stream_reader.rs

```diff
@@ -4,17 +4,22 @@ use std::cmp::{max, min};
 use std::sync::Arc;
 use std::time;
 
-use futures::StreamExt;
 use itertools::{repeat_n, Itertools};
 use scylla::frame::response::result::CqlValue;
 use scylla::Session;
+use scylla::transport::errors::QueryError;
 use tokio::sync::watch;
 use tokio::time::sleep;
 
 use crate::cdc_types::{GenerationTimestamp, StreamID};
 use crate::checkpoints::{start_saving_checkpoints, CDCCheckpointSaver, Checkpoint};
 use crate::consumer::{CDCRow, CDCRowSchema, Consumer};
 
+const BASIC_TIMEOUT_SLEEP: u64 = 10;
+const TIMEOUT_FACTOR: u64 = 2;
+const TIMEOUT_LIMIT: u64 = 30_000;
```

Comment:
Let's not hardcode the `TIMEOUT_LIMIT`. I think we can just use `sleep_interval` as a cap.